src/MediatR.Extensions.Autofac.DependencyInjection/Builder/MediatRConfigurationBuilder.cs

```diff
@@ -152,10 +157,16 @@ public MediatRConfigurationBuilder WithCustomStreamPipelineBehaviors(IEnumerable
             this.internalOpenGenericHandlerTypesToRegister.ToArray(),
             this.internalCustomPipelineBehaviorTypes.ToArray(),
             this.internalCustomStreamPipelineBehaviorTypes.ToArray(),
-            this.registrationScope);
+            this.registrationScope, this.internalCustomRegistrationAction);
     
     private void AddOpenGenericHandlerToRegister(Type openHandlerType)
     {
         this.internalOpenGenericHandlerTypesToRegister.Add(openHandlerType);
     }
+
+    public MediatRConfigurationBuilder WithOpenGenericHandlerTypeToRegisterCallback(Action<IRegistrationBuilder<object, ScanningActivatorData, DynamicRegistrationStyle>> openGenericHandlerTypeToRegisterCallback)
```

Comment:
Please add tests for the new functionality so that we may validate that a callback is actually executed.